File: SRC/Dart/Hanoi.dart
import 'dart:io';

//하노이의 탑

void hanoi(n, from, to, via) {
  if(n > 1) {
    
    hanoi(n - 1, from, via, to);
    print(from + " -> " + to);
    hanoi(n - 1, via, to, from);

  } else {
    print(from + " -> " + to);
  }
}

void main() {
  var n = int.parse(stdin.readLineSync());
  hanoi(n, "a", "b", "c");
}

File: SRC/C#/Hanoi.cs
using System;

//하노이의 탑
public class Hanoi
{
    private void hanoi(int n, char from, char to, char via)
    {
        if(n > 1) {
            hanoi(n - 1, from, via, to);
            Console.WriteLine(from + " -> " + to);
            hanoi(n - 1, via, to, from);
        } else {
            Console.WriteLine(from + " -> " + to);
        }
    }

    public static void Main() {
        int n = int.Parse(Console.ReadLine());
        Hanoi h = new Hanoi();
        h.hanoi(n, 'a', 'b', 'c');
    }

}
